worlds/ffmq/Items.py

```diff
@@ -0,0 +1,295 @@
+    "Pazuzu 2F": ItemData(None, ItemClassification.progression),
+    "Pazuzu 3F": ItemData(None, ItemClassification.progression),
+    "Pazuzu 4F": ItemData(None, ItemClassification.progression),
+    "Pazuzu 5F": ItemData(None, ItemClassification.progression),
+    "Pazuzu 6F": ItemData(None, ItemClassification.progression),
+    "Dark King": ItemData(None, ItemClassification.progression),
+    #"Barred": ItemData(None, ItemClassification.progression),
+
+}
+
+prog_map = {
+    "Swords": "Progressive Sword",
+    "Axes": "Progressive Axe",
+    "Claws": "Progressive Claw",
+    "Bombs": "Progressive Bomb",
+    "Shields": "Progressive Shield",
+    "Armors": "Progressive Armor",
+    "Helms": "Progressive Helm",
+    "Accessories": "Progressive Accessory",
+}
+
+
+def yaml_item(text):
+    if text == "CaptainCap":
+        return "Captain's Cap"
+    elif text == "WakeWater":
+        return "Wakewater"
+    return "".join(
+        [(" " + c if (c.isupper() or c.isnumeric()) and not (text[i - 1].isnumeric() and c == "F") else c) for
+         i, c in enumerate(text)]).strip()
+
+
+item_groups = {}
+for item, data in item_table.items():
+    for group in data.groups:
+        item_groups[group] = item_groups.get(group, []) + [item]
+
+
+def create_items(self) -> None:
+    items = []
+    starting_weapon = self.multiworld.starting_weapon[self.player].current_key.title().replace("_", " ")
+    if self.multiworld.progressive_gear[self.player]:
+        for item_group in prog_map:
+            if starting_weapon in self.item_name_groups[item_group]:
+                starting_weapon = prog_map[item_group]
+                break
+    self.multiworld.push_precollected(self.create_item(starting_weapon))
+    self.multiworld.push_precollected(self.create_item("Steel Armor"))
+    if self.multiworld.sky_coin_mode[self.player] == "start_with":
+        self.multiworld.push_precollected(self.create_item("Sky Coin"))
+
+    def add_item(item_name):
+        if item_name in ["Steel Armor", "Sky Fragment"] or "Progressive" in item_name:
+            return
+        if item_name.lower().replace(" ", "_") == self.multiworld.starting_weapon[self.player].current_key:
+            return
+        if self.multiworld.progressive_gear[self.player]:
+            for item_group in prog_map:
+                if item_name in self.item_name_groups[item_group]:
+                    item_name = prog_map[item_group]
+                    break
+        if item_name == "Sky Coin":
+            if self.multiworld.sky_coin_mode[self.player] == "shattered_sky_coin":
+                for _ in range(40):
+                    items.append(self.create_item("Sky Fragment"))
+                return
+            elif self.multiworld.sky_coin_mode[self.player] == "save_the_crystals":
+                items.append(self.create_filler())
+                return
+        i = self.create_item(item_name)
+        if i in self.multiworld.precollected_items:
```

Comment:
It might be faster to built a lookup of all the precollected item names and just reference that near the beginning of this function instead of creating the item?

Reply:
Done